manager/main.go

```diff
@@ -56,9 +57,20 @@ func init() {
 func run(namespace string, metricsAddr string, enableLeaderElection bool,
 	enableApplicationController, enableBlueprintController, enablePlotterController, enableMotionController bool) int {
 	setupLog.Info("creating manager")
+
+	var applicationNamespaceSelector fields.Selector
+	applicationNamespace := utils.GetApplicationNamespace()
+	if len(applicationNamespace) > 0 {
+		applicationNamespaceSelector = fields.SelectorFromSet(fields.Set{"metadata.namespace": applicationNamespace})
+	}
+	setupLog.Info("Application namespace: " + applicationNamespace)
+
+	blueprintNamespace := utils.GetBlueprintNamespace()
+
 	systemNamespaceSelector := fields.SelectorFromSet(fields.Set{"metadata.namespace": utils.GetSystemNamespace()})
-	workerNamespaceSelector := fields.SelectorFromSet(fields.Set{"metadata.namespace": app.BlueprintNamespace})
+	workerNamespaceSelector := fields.SelectorFromSet(fields.Set{"metadata.namespace": blueprintNamespace})
 	selectorsByObject := cache.SelectorsByObject{
+		&appv1.FybrikApplication{}:      {Field: applicationNamespaceSelector},
```

Comment:
Can applicationNamespaceSelector be added on condition rather than adding an empty selector when not required?

Reply:
Is there a problem adding an empty selector?

Reply:
We can leave as-is for now